lib/base64.c

```diff
@@ -224,27 +217,30 @@ static CURLcode base64_encode(const char *table64,
 
     switch(inputparts) {
     case 1: /* only one byte read */
-      msnprintf(output, 5, "%c%c==",
-                table64[obuf[0]],
-                table64[obuf[1]]);
+      i = msnprintf(output, 5, "%c%c%s%s",
+                    table64[obuf[0]],
+                    table64[obuf[1]],
+                    padstr,
+                    padstr);
```

Comment:
I don't see how using `table64 + 64` is better, if anything it's harder to understand.

Reply:
In the base64url case, we have a padding string of length 0, in base 64, this is "=" of length 1.
We thus cannot assume padding is a single character: we must use a string. Having the latter appended to the char table itself is the simplest way to achieve it. It is much more readable to have `padstr` rather than `table64 + 64` in the calls, and probably optimized by the compiler.

Reply:
Ok I think I understand now. There's two base64 encodings, the first regular, 
"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
the second for url
"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_"
and in the latter case padstr (table+64) is going to be null?